src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -276,6 +277,42 @@ public TimeSpan Expect100ContinueTimeout
         public IDictionary<string, object?> Properties =>
             _settings._properties ?? (_settings._properties = new Dictionary<string, object?>());
 
+        /// <summary>
+        /// Represents a method that specifies the <see cref="Encoding"/> to use when interpreting header values.
+        /// </summary>
+        /// <param name="headerName">Name of the header to specify the <see cref="Encoding"/> for.</param>
+        /// <param name="request">The <see cref="HttpRequestMessage"/> we are enoding/decoding the headers for.</param>
+        /// <returns><see cref="Encoding"/> to use or <see langword="null"/> to use the default behavior.</returns>
+        public delegate Encoding? HeaderEncodingSelector(string headerName, HttpRequestMessage request);
```

Comment:
The delegate is part of the class and not the namespace? I'm not sure how often we do that in BCL. @bartonjs thoughts?

Reply:
It should be in the namespace, not nested in the class.

Reply:
I listed some of the API detail questions in the PR description.

If it is moved to the namespace level, should the delegate be
`public delegate Encoding? HeaderEncodingSelector<TContext>(string headerName, TContext context);`
Wtih SocketsHttpHandler implementing `HeaderEncodingSelector<HttpRequestMessage>` and MultipartContent implementing `HeaderEncodingSelector<HttpContent>` as suggested in API review?

Should it be in the `System.Net.Http` or `System.Net.Http.Headers` namespace?

Reply:
System.Net.Http

Reply:
Changed to `HeaderEncodingSelector<TContext>(string headerName, TContext context)` in `System.Net.Http` namespace